test/hpr_routing_SUITE.erl

```diff
@@ -118,5 +160,110 @@ join_req_test(_Config) ->
 
     ?assert(meck:validate(hpr_protocol_router)),
     meck:unload(hpr_protocol_router),
+    ok.
+
+max_copies_test(_Config) ->
+    MaxCopies = 2,
+    DevAddr = 16#00000000,
+    {ok, NetID} = lora_subnet:parse_netid(DevAddr, big),
+    Route = hpr_route:new(#{
+        net_id => NetID,
+        devaddr_ranges => [#{start_addr => 16#00000000, end_addr => 16#0000000A}],
+        euis => [#{app_eui => 1, dev_eui => 1}, #{app_eui => 1, dev_eui => 2}],
+        oui => 1,
+        server => #{
+            host => <<"127.0.0.1">>,
+            port => 80,
+            protocol => {packet_router, #{}}
+        },
+        max_copies => MaxCopies
+    }),
+    ok = hpr_config:insert_route(Route),
+
+    meck:new(hpr_protocol_router, [passthrough]),
+    meck:expect(hpr_protocol_router, send, fun(_, _, _) -> ok end),
+
+    #{secret := PrivKey1, public := PubKey1} = libp2p_crypto:generate_keys(ecc_compact),
+    SigFun1 = libp2p_crypto:mk_sig_fun(PrivKey1),
+    Gateway1 = libp2p_crypto:pubkey_to_bin(PubKey1),
+
+    UplinkPacketUp1 = test_utils:uplink_packet_up(#{
+        gateway => Gateway1, sig_fun => SigFun1, devaddr => DevAddr, fcnt => 1
+    }),
+
+    #{secret := PrivKey2, public := PubKey2} = libp2p_crypto:generate_keys(ecc_compact),
+    SigFun2 = libp2p_crypto:mk_sig_fun(PrivKey2),
+    Gateway2 = libp2p_crypto:pubkey_to_bin(PubKey2),
+
+    UplinkPacketUp2 = test_utils:uplink_packet_up(#{
+        gateway => Gateway2, sig_fun => SigFun2, devaddr => DevAddr, fcnt => 1
+    }),
+
+    #{secret := PrivKey3, public := PubKey3} = libp2p_crypto:generate_keys(ecc_compact),
```

Comment:
Maybe create a utility function to make a up link packet, to avoid this sequence of code. Might be a bit clearer what the test is doing.